nixos/modules/services/web-apps/healthchecks.nix

```diff
@@ -0,0 +1,216 @@
+      default = "[::1]";
+      description = ''
+        Address the server will listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8000;
+      description = ''
+        Port the server will listen on.
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/healthchecks";
+      description = ''
+        Storage path of healthchecks.
+      '';
+    };
+
+    secretKeyFile = mkOption {
+      type = types.path;
+      description = ''
+        Path to a file containing the secret key.
+      '';
+    };
+
+    debug = mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Enable debug mode.
+      '';
+    };
+
+    registrationOpen = mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        A boolean that controls whether site visitors can create new accounts. Set it to False if you are setting up a private Healthchecks instance, but it needs to be publicly accessible (so, for example, your cloud services can send pings to it).
+        If you close new user registration, you can still selectively invite users to your team account.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Additional lines of configuration appended to the <literal>local_settings.py</literal>.
+        See the <link xlink:href="https://healthchecks.io/docs/self_hosted_configuration/">documentation</link> for more possible options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ healthchecksManageScript ];
+
+    systemd.targets.healthchecks = {
+      description = "Target for all Healthchecks services";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target"];
+    };
+
+    systemd.services = let
+      defaultServiceConfig = {
+        WorkingDirectory = "${cfg.dataDir}";
+        User = "healthchecks";
+        Group = "healthchecks";
+        StateDirectory = "healthchecks";
+        StateDirectoryMode = "0750";
+        Restart = "on-failure";
+      };
+    in {
+      healthchecks-migration = {
+        description = "Healthchecks migrations";
+        wantedBy = [ "healthchecks.target" ];
+
+        environment = {
+          PYTHONPATH = pkg.pythonPath;
+        };
+
+        serviceConfig = defaultServiceConfig // {
+          Type = "oneshot";
+          ExecStart = ''
+            ${pkg}/bin/healthchecks migrate
+          '';
+        };
+      };
+
+      healthchecks = {
+        description = "Healthchecks WSGI Service";
+        wantedBy = [ "healthchecks.target" ];
+        after = [ "healthchecks-migration.service" ];
+
+        preStart = ''
+          ${pkg}/bin/healthchecks collectstatic --no-input
+          ${pkg}/bin/healthchecks remove_stale_contenttypes --no-input
+          ${pkg}/bin/healthchecks compress
```

Comment:
That should be done at compile time in the package to only run it once.
Doesn't this also write static files somewhere which would be lost?

Reply:
But its behavior depends on settings which could be modified via `extraConfig`, so i think packages build-time is the wrong place to run it?

It writes files to STATIC_ROOT which is set to /var/lib/healthchecks/static by default, so should be persistent on most systems.